projects/ovrball/ovrball.c

```diff
@@ -668,17 +669,48 @@ void example_scene_setup( void * tag, void * opaquev )
 	paddlesolid = CNOVRModelCreate( 0, GL_TRIANGLES );
 	CNOVRModelLoadFromFileAsync( paddlesolid, "paddle.obj:barytc" );
 
-	pose_make_identity( &paddetransform );
+	pose_make_identity( &paddetransform1 );
+	pose_make_identity( &paddetransform2 );
+	
+	//Default back to knuckles if nothing is found
+	//Copy knuckles values in cnovr_quat
+	quatcopy( paddetransform1.Rot, &( paddle_angles[0] ) );
+	quatcopy( paddetransform2.Rot, paddetransform1.Rot );
+	
+	//Check every devices tracked
+	for( i = 0; i < MAX_POSES_TO_PULL_FROM_OPENVR; ++i )
+	{
+		rv = CNOVRGetTrackedDeviceString( i, ETrackedDeviceProperty_Prop_ModelNumber_String );
+		
+		if( strlen(rv) <= 0 ) break;	//We reached the end of valid devices, rv <= 0
+		
+		is_vive = strstr( rv, "Vive" )?1:0;
+		is_controller = strstr( rv, "Controller" )?1:0;
+
+		//Are we using something that is a Vive?
+		if( is_vive == 1 )
```

Comment:
Please use proper {'s around all code blocks, since it is unclear where the blocks of code are if an `else` statement is used.